mantidproject/mantid
# New ConvertHFIRSCDtoMDE algorithm for WAND and DEMAND

**Description of work.**

This algorithm has the same functionality as [ConvertWANDSCDtoQ](https://docs.mantidproject.org/nightly/algorithms/ConvertWANDSCDtoQ-v1.html) except that it convert to a MDEventWorkspace instead of MDHistoWorkspace, and only to Q_sample with no normalization.


The advantage of using an MDEventWorkpsace is it should provided more acturate results from FindPeaksMD and will also enable the use of Peak integration algorithm in Mantid that only work for MDEventWorksapces, like [IntegratePeaksMD](https://docs.mantidproject.org/nightly/algorithms/IntegratePeaksMD-v2.html)



**To test:**

This should work with both WAND and DEMAND data. View the workspaces in sliceviewer and check that they look correct.

WAND example, you can also try the usage example for another WAND test data set.
```python
data = LoadWANDSCD(IPTS=22745, RunNumbers='147131-148931', Grouping='4x4')
mde=ConvertHFIRSCDtoMDE(data, Wavelength=1.488)
```

DEMAND example, you should get the same results as in #27536
```python
data2 = Load("/HFIR/HB3A/IPTS-24655/shared/autoreduce/HB3A_exp0724_scan0183.nxs")
mde2=ConvertHFIRSCDtoMDE(data2, Wavelength=1.008)
```

*There are no issues.*


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
